nexus/src/app/update.rs

```diff
@@ -276,4 +304,295 @@ impl super::Nexus {
         })?;
         Ok(body)
     }
+
+    pub async fn system_update_create(
+        &self,
+        opctx: &OpContext,
+        create_update: CreateSystemUpdate,
+    ) -> CreateResult<db::model::SystemUpdate> {
+        let now = Utc::now();
+        let update = db::model::SystemUpdate {
+            identity: db::model::SystemUpdateIdentity {
+                id: Uuid::new_v4(),
+                time_created: now,
+                time_modified: now,
+            },
+            version: SemverVersion(create_update.version),
+        };
+        self.db_datastore.system_update_create(opctx, update).await
+    }
+
+    pub async fn component_update_create(
+        &self,
+        opctx: &OpContext,
+        create_update: CreateComponentUpdate,
+    ) -> CreateResult<db::model::ComponentUpdate> {
+        let now = Utc::now();
+        let update = db::model::ComponentUpdate {
+            identity: db::model::ComponentUpdateIdentity {
+                id: Uuid::new_v4(),
+                time_created: now,
+                time_modified: now,
+            },
+            version: SemverVersion(create_update.version),
+            component_type: create_update.component_type,
+            parent_id: create_update.parent_id,
+        };
+
+        // TODO: make sure system update with that ID exists first
+
+        self.db_datastore
+            .component_update_create(
+                opctx,
+                create_update.system_update_id,
+                update,
+            )
+            .await
+    }
+
+    pub async fn system_update_fetch_by_id(
+        &self,
+        opctx: &OpContext,
+        update_id: &Uuid,
+    ) -> LookupResult<db::model::SystemUpdate> {
+        // TODO: I don't think this is the right way to do this auth check
+        opctx.authorize(authz::Action::ListChildren, &authz::FLEET).await?;
```

Comment:
For this, I don't _think_ you need an explicit authorize call. When you call `fetch()` below that checks to ensure you have read access to the `SystemUpdate` resource. That's all you should really need here. The `ListChildren` of the parent is only required on a list endpoint.

Reply:
I thought that, but when I didn't do this, my authorization test showed it was being more permissive than I expected. That could mean I gave the wrong arguments to the auth resource macro.